dask_expr/merge.py

```diff
@@ -0,0 +1,173 @@
+import functools
+
+from dask.dataframe.dispatch import make_meta, meta_nonempty
+from dask.utils import M, apply
+
+from dask_expr.expr import Blockwise, Expr
+from dask_expr.repartition import Repartition
+from dask_expr.shuffle import Shuffle, _contains_index_name
+
+
+class Merge(Expr):
+    """Abstract merge operation"""
```

Comment:
Maybe instead 
```suggestion
    """Merge / join two dataframes 
    
    This is an abstract class.  It will be transformed into a concrete 
    implementation before graph construction.
    
    See Also
    --------
    BlockwiseMerge
    Shuffle
    """
```